packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -647,6 +651,309 @@ describeWithDOM('mount', () => {
 </Foo>`);
     });
   });
+  describeIf(is('>= 16.8'), 'hooks', () => {
+    describe('useState', () => {
+
+      /*
+        Test Component
+      */
+      function FooCounter({ initialCount = 0 }){
+        const [count, setCount] = useState(+initialCount);
+
+        return (
+          <Fragment>
+              <button className="increment" type="button" onClick={() => setCount( count + 1)}>-</button>
+                <span className="counter">
+                  {count}
+                </span>
+              <button className="decrement" type="button" onClick={() => setCount( count - 1)}>+</button>
+          </Fragment>
+        );
+      };
+      
+      // Test data
+      let wrapper;
+      const initialCount = 5;
+
+      before(() => {
```

Comment:
please use `beforeEach`, and never `before` - but instead, it'd be better to copy-paste this wrapper initialization into each `it`. DRY isn't often a great principle in tests.

Reply:
Discussion:
@ljharb : I have added one describe as one test case which has series of effects which needs to be tested where remounting of the component is actually involved in that test case, thats why you see I added different describe for all different scenarios/test case. Please leme know if I am missing/ need more info.

Reply:
In that case I'd prefer a single `it` to be responsible for mount, unmount, and remount, rather than making the ordering of `it`s important.

Reply:
Sure, I will make those changes

Reply:
Done